app/js/components/manageApps/Addon.jsx

```diff
@@ -83,6 +90,71 @@ class Addon extends Component {
 
   }
 
+  fetchCurrentUser(){
+    const apiHelper = new ApiHelper(null);
+    const getUserData = new Promise(function(resolve, reject) {
+      apiHelper.get('/v1/session').then(response => {
+        resolve(response);
+      });
+    });
+    return getUserData;
+  }
+
+  checkIsAdmin() {
+    this.fetchCurrentUser().then((response) => {
+      if (response.user) {
+        const user = response.user;
+        let userPrivileges = [];
+        let userCanManageModules;
+        let userHasPermittedRoles;
+        userPrivileges = userPrivileges.concat(user.privileges);
+        if (user.roles.length > 0){
```

Comment:
What happens if the user object has no roles key? Some browsers may throw errors.

Saying `if(user.roles & user.roles.length>0)` may be better

Reply:
if I do this it adds a lot of code which will be redundant and also I will be repeating the same check twice since all am checking is if the array has content so it can proceed.